File: legacy/killFlats.pl
: # Use perl
   eval 'exec perl -S $0 "$@"'
   if 0;

# it is faster to use something like
#!/usr/local/bin/perl5 
# but this is more portable.

# expects a file with points (one per line)
# and drops the parts that are within log(2) from
# the top and the bottom of the plot
# It is supposed to replace 'choptop' and 'chopbottom'
# simultaneously

$verbose = 0;
$debug = 0;


while( $_ = $ARGV[0], /^-/){
   shift;
   last if /^--$/;
   if( /^-D(.*)/){ $debug = $1 }
   if( /^-k(.*)/){ $k = $1 }
   if (/^-v/)    { $verbose++ }
}

$nargs = $#ARGV + 1;   # the number of remaining arguments

if( $nargs > 1 ) { die "USAGE: $0 <file>"  }

# print "first filename is=", $ARGV[0], "\n";


$HUGE = 2^30;
$miny = $HUGE;
$maxy = -$HUGE;
$width = log(2); #drop all points with $y within $width from min or max

$lineno = 0;
while(<>){
    # chomp;
    # s/\s//g; # kill blanks
    # print $_, "\n";
    $lineno ++;
    ($x, $y) = split;
    if($verbose)  { print $x, " ", $y, "\n"; }
    $xa[$lineno] = $x;
    $ya[$lineno] = $y;

    if( $y < $miny) { $miny = $y;}
    if ( $y > $maxy) { $maxy = $y ; }
}

$totpts = $lineno;

if( $verbose ) { print "maxy=", $maxy, " miny=", $miny, "\n" ;}

for( $i=1; $i<= $totpts; $i++){
    $y = $ya[$i];
    if( ( $y > $miny + $width ) && ($y < $maxy - $width) ) {
      print $xa[$i], " ", $ya[$i], "\n";
    }
}

File: fdnq/killFlats.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
expects an array with points and drops the parts that are within log(2) from
the top and the bottom of the plot
It is supposed to replace 'choptop' and 'chopbottom' simultaneously
"""
__author__ = 'Shubhranshu-Shekhar'
__date__ = 'July, 2020'

import math


def killFlats(box_count_arr=None, verbose=0):
    flat_arr = []

    HUGE = 2 ^ 30
    miny = HUGE
    maxy = -HUGE
    width = math.log(2)  # drop all points with $y within $width from min or max

    lineno = 0
    xa, ya = {}, {}
    for line in box_count_arr: #  fileinput.input():  #
        lineno += 1
        # x, y = line.strip().split()
        x, y = line
        x = float(x)
        y = float(y)

        if verbose > 0:
            print(x, " ", y)

        xa[lineno] = x
        ya[lineno] = y

        if y < miny:
            miny = y

        if y > maxy:
            maxy = y

    totpts = lineno

    if verbose > 0:
        print("maxy=", maxy, " miny=", miny)

    for i in range(1, totpts+1):
        y = ya[i]
        if (y > (miny + width)) and (y < (maxy - width)):
            # print(xa[i], " ", ya[i])
            flat_arr.append((xa[i], ya[i]))

    return flat_arr


# if __name__ == '__main__':
#     killFlats()
